core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -938,17 +986,17 @@ private void initAfterRebalance() {
                 initAfterRebalance();
             }
 
-            ByteBuffer bb = state.getChunk().readKey(state.getIndex()).slice();
-            Handle currentHandle = state.getChunk().getHandle(state.getIndex());
+            ByteBuffer bb = needsKey ? state.getChunk().readKey(state.getIndex()).slice() : null;
+            Slice currentValue = needsValue ? state.getChunk().getValueSlice(state.getIndex()) : null;
```

Comment:
If value is deleted getValueSlice will return null. Isn't it a problem? Didn't we say we need a loop here?

Reply:
I now added the loop, but there is still the problem of `hasNext` being incorrect because of this loop.